netty/src/main/java/io/grpc/netty/NettyClientHandler.java

```diff
@@ -472,7 +473,11 @@ private void sendPingFrame(ChannelHandlerContext ctx, SendPingCommand msg,
       @Override
       public void operationComplete(ChannelFuture future) throws Exception {
         if (!future.isSuccess()) {
-          finalPing.failed(future.cause());
+          Throwable cause = future.cause();
+          if (cause instanceof ClosedChannelException) {
+            cause = lifecycleManager.getShutdownThrowable();
```

Comment:
I don't believe PingCallback was intended to allow a null cause. And such a null cause is very difficult to debug.

We should just create a status in that case, [like done elsewhere](https://github.com/grpc/grpc-java/blob/master/netty/src/main/java/io/grpc/netty/NettyClientTransport.java#L244), maybe with the message "Ping failed but for unknown reason".